### PR TITLE
New models for UZIS data, migrations, app refactoring

### LINKED ISSUE #56: Add vacc. places from UZIS opendata
Add to our DB official dataset from UZIS with off. vacc. places and link them to our ockovaci_mista

### PR DESCRIPTION
**Pull request by jan-stanek**
closes #56, #57, #58

TODO:
- [x] test/fix fetcher
- [x] configure flask to run on 0.0.0.0
- [x] create google table with additional ids
- [x] add fetcher for the google table 
- [x] fix views after the last refactoring (VolnaMistaRaw->VolnaMistaDen, VolnaMista->VolnaMistaCas)
- [x] set log level to info
- [x] implement API check in google table
- [x] fill google table
- [x] fix CSV exports

Upgrade guide:
- drop all tables in database (or create new database)
- check connection string in config.py
- install new requirements
- execute database migrations: `flask db upgrade` 
- configure fetchers
  - fill config.ini according to the template
  - create google_key.json in the root directory (documented here: https://gspread.readthedocs.io/en/latest/oauth2.html)
- execute fetchers: `flask fetch-all`
- execute webserver `flask run --host 0.0.0.0`
- publish is without changes

### CODE REVIEW

**Comment by jan-stanek** [comment]
`flask run --host 0.0.0.0` can be used instead of hardcoded config

**Comment by jan-stanek** [comment]
This can be used for checking links in google sheet: https://banhawy.medium.com/how-to-use-google-spreadsheets-to-check-for-broken-links-1bb0b35c8525

File: app/commands.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@app.cli.command('fetch-reservatic')
+def fetch_reservatic_command():
+    """Fetch free capacities from Reservatic."""
+    fetcher = ReservaticFetcher()
+    fetcher.fetch_free_capacities()
+
+@app.cli.command('fetch-opendata')
+def fetch_opendata_command():
+    """Fetch opendata from UZIS."""
+    fetcher = OpendataFetcher()
```
Comment (msusicky) [inline app/commands.py:16]: OpenData will be better

File: app/sheet_fetcher.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import configparser
+
+import gspread as gspread
+
+from app import db
+from app.models import OckovaciMisto
+
+
+class SheetFetcher:
+    SHEET_API = 'https://onemocneni-aktualne.mzcr.cz/api/v2/covid-19/prehled-ockovacich-mist.json'
```
Comment (msusicky) [inline app/sheet_fetcher.py:10]: Maybe these constants should be all in one place ?

File: app/views.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        .filter(
+        VolnaMistaDen.import_id == last_update_import_id()).order_by(
+        VolnaMistaDen.datum, OckovaciMisto.okres, OckovaciMisto.nazev).all()
+
+    return render_template('kraj.html', data=nactene_informace, kraj=kraj, last_update=last_update())
+
+
+@bp.route("/misto/<misto>")
+def info_misto(misto):
+    nactene_informace = db.session.query(Okres.nazev.label("okres"), Kraj.nazev.label("kraj"), OckovaciMisto.nazev,
```
Comment (msusicky) [inline app/views.py:58]: Please add if misto is none catch

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
+Flask-Migrate
 SQLAlchemy>=1.0.2
 Werkzeug>=0.16.1
 psycopg2==2.7.7
+python-dotenv
 markupsafe
-requests
\ No newline at end of file
+requests
+gspread
```
Comment (msusicky) [inline requirements.txt:11]: Please add next time requested actions -> pip should be executed again :)

**Review by msusicky — CHANGES_REQUESTED** [review]
My comments below -> I've tested it, looks good but small changes are necessary.
+ Please add upgrade guide to the Pull request notes, because it is a huge move forward.
Thanks

File: app/commands.py (hunk 1 above)
Comment (jan-stanek) [reply app/commands.py:16]: renamed

File: app/sheet_fetcher.py (hunk 2 above)
Comment (jan-stanek) [reply app/sheet_fetcher.py:10]: I would leave it for improvements, count of days should be also somewhere else, but it has to be reflected also in GUI

File: requirements.txt (hunk 4 above)
Comment (jan-stanek) [reply requirements.txt:11]: sorry :)
we should update also setup.py, or remove it

File: app/views.py (hunk 3 above)
Comment (jan-stanek) [reply app/views.py:62]: added

**Review by msusicky — APPROVED** [review]

**Pull request merged**